spec/policy/http_proxy/http_proxy_spec.lua

```diff
@@ -0,0 +1,53 @@
+local proxy_policy = require('apicast.policy.http_proxy')
+local resty_url = require 'resty.url'
+
+describe('HTTP proxy  policy', function()
+  local all_proxy_val = "http://all.com"
+  local http_proxy_val = "http://plain.com"
+  local https_proxy_val = "http://secure.com"
+
+  it("http[s] proxies are defined if all_proxy is in there", function()
+    local proxy = proxy_policy.new({
+      all_proxy = all_proxy_val
+    })
+    assert.same(proxy:find_proxy("http"), resty_url.parse(all_proxy_val))
```

Comment:
In all these tests, I think we should check what `export` returns instead of `find_proxy`, because `export` is what should be exposed to other modules.